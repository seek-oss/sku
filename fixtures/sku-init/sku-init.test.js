const path = require('node:path');
const fs = require('node:fs/promises');
const { promisify } = require('node:util');
const exec = promisify(require('node:child_process').exec);
const { runSkuScriptInDir } = require('@sku-private/test-utils');

const fixtureDirectory = __dirname;
const projectName = 'new-project';
const projectDirectory = path.join(fixtureDirectory, projectName);

describe('sku init', () => {
  beforeAll(async () => {
    await fs.rm(projectDirectory, { recursive: true, force: true });
  });

  afterAll(async () => {
    await fs.rm(projectDirectory, { recursive: true, force: true });

    console.log(
      "Running 'pnpm install' to clean up lockfile after sku-init test...",
    );
    await exec('pnpm install');
    console.log('Cleanup complete');
  });

  it(
    'should create a sku.config.ts',
    async () => {
      const projectName = 'new-project';
      await fs.rm(path.join(fixtureDirectory, projectName), {
        recursive: true,
        force: true,
      });

      const { child, stdout, stderr } = await runSkuScriptInDir(
        'init',
        fixtureDirectory,
        [projectName],
      );

      console.log('sku init stdout');
      console.log(stdout);
      console.log('sku init stderr');
      console.error(stderr);

      expect(child.exitCode).toBe(0);

      const skuConfig = await fs.readFile(
        path.join(fixtureDirectory, projectName, 'sku.config.ts'),
        'utf-8',
      );

      expect(skuConfig).toMatchInlineSnapshot(`
"import type { SkuConfig } from 'sku';

const skuConfig: SkuConfig = {
  clientEntry: 'src/client.tsx',
  renderEntry: 'src/render.tsx',
  environments: ['development', 'production'],
  publicPath: '/path/to/public/assets/', // <-- Required for sku build output
};

export default skuConfig;
"
`);
    },
    // `sku init` is a long running task and can take some time to complete
    5 * 60 * 1000,
  );
});
