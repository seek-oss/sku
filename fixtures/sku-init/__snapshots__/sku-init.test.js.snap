// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sku init should create .gitignore 1`] = `
"node_modules
npm-debug.log

# managed by sku
.eslintcache
.prettierrc
coverage/
dist/
eslint.config.js
report/
tsconfig.json
# end managed by sku
"
`;

exports[`sku init should create .prettierignore 1`] = `
"# managed by sku
.eslintcache
.prettierrc
coverage/
dist/
eslint.config.js
pnpm-lock.yaml
report/
tsconfig.json
# end managed by sku
"
`;

exports[`sku init should create README.md 1`] = `
"# new-project

This project is powered by [sku](https://github.com/seek-oss/sku), [braid](https://github.com/seek-oss/braid-design-system) and built with [React](https://facebook.github.io/react).

## Getting Started

First of all, make sure you&#39;ve installed [pnpm](https://pnpm.io/installation).

Then, install dependencies:

\`\`\`sh
$ pnpm i
\`\`\`

## Workflow

Start a local development server:

\`\`\`sh
$ pnpm run start
\`\`\`

Run unit tests:

\`\`\`sh
$ pnpm run test
\`\`\`

Lint and format code:

\`\`\`sh
$ pnpm run lint
$ pnpm run format
\`\`\`

Build assets for production:

\`\`\`sh
$ pnpm run build
\`\`\`
"
`;

exports[`sku init should create eslint.config.js 1`] = `
"/** THIS FILE IS GENERATED BY SKU, MANUAL CHANGES WILL BE DISCARDED **/
const { eslintConfigSku } = require('sku/config/eslint');

module.exports = eslintConfigSku;"
`;

exports[`sku init should create package.json 1`] = `
{
  "dependencies": {
    "braid-design-system": "VERSION_IGNORED",
    "react": "VERSION_IGNORED",
    "react-dom": "VERSION_IGNORED",
  },
  "devDependencies": {
    "@types/react": "VERSION_IGNORED",
    "@types/react-dom": "VERSION_IGNORED",
    "@vanilla-extract/css": "VERSION_IGNORED",
    "sku": "VERSION_IGNORED",
  },
  "name": "new-project",
  "private": true,
  "scripts": {
    "build": "sku build",
    "format": "sku format",
    "lint": "sku lint",
    "serve": "sku serve",
    "start": "sku start",
    "test": "sku test",
  },
  "version": "0.1.0",
}
`;

exports[`sku init should create sku.config.ts 1`] = `
"import type { SkuConfig } from 'sku';

const skuConfig: SkuConfig = {
  clientEntry: 'src/client.tsx',
  renderEntry: 'src/render.tsx',
  environments: ['development', 'production'],
  publicPath: '/path/to/public/assets/', // <-- Required for sku build output
};

export default skuConfig;
"
`;

exports[`sku init should create src/App/NextSteps.tsx 1`] = `
"import {
  Box,
  Heading,
  Text,
  Strong,
  ContentBlock,
  TextLink,
  Stack,
  List,
  Divider,
} from 'braid-design-system';

import * as styles from './NextSteps.css';

interface NextStepsProps {
  environment: 'development' | 'production';
}
export function NextSteps({ environment }: NextStepsProps) {
  return (
    <Box
      paddingY="xxlarge"
      paddingX={{ mobile: 'small', tablet: 'gutter' }}
      background="customDark"
      className={styles.background}
    >
      <ContentBlock width="large">
        <Stack space="xxlarge">
          <Stack space="xlarge">
            <Stack space="medium" align="center">
              <Heading level="1" component="span">
                üéâ
              </Heading>
              <Heading level="1" weight="weak">
                Congratulations!
              </Heading>
            </Stack>
            <Stack space="medium" align="center">
              <Text size="large">
                <Strong>{'new-project'}</Strong> has been successfully
                initialised.
              </Text>
              <Text>Current environment: {environment}</Text>
            </Stack>
          </Stack>

          <Box
            background="surface"
            borderRadius="xlarge"
            boxShadow="large"
            padding={{ mobile: 'large', tablet: 'xlarge', desktop: 'xxlarge' }}
          >
            <Stack space={{ mobile: 'xlarge', desktop: 'xxlarge' }}>
              <Stack space="large">
                <Heading level="2">üß∞ Features and Tooling</Heading>
                <Text>
                  The project comes pre-configured with a lot of standardised
                  tooling to make authoring web applications easier.
                </Text>

                <Stack space="gutter">
                  <Heading level="3">Platform</Heading>

                  <Text>
                    The{' '}
                    <TextLink href="https://seek-oss.github.io/sku/#/">
                      sku platform
                    </TextLink>{' '}
                    provides a number of tools to make development more
                    efficient, for example,{' '}
                    <TextLink href="https://www.typescriptlang.org/">
                      TypeScript
                    </TextLink>
                    , <TextLink href="https://jestjs.io/">Jest</TextLink>,{' '}
                    <TextLink href="https://eslint.org/">ESLint</TextLink>, and
                    more.
                  </Text>
                </Stack>

                <Stack space="gutter">
                  <Heading level="3">Components</Heading>

                  <Text>
                    The project comes with{' '}
                    <TextLink href="https://seek-oss.github.io/braid-design-system/">
                      Braid Design System
                    </TextLink>{' '}
                    ‚Äî SEEK&rsquo;s themeable component library.
                  </Text>
                </Stack>

                <Stack space="gutter">
                  <Heading level="3">Translations</Heading>

                  <Text>
                    The project comes with{' '}
                    <TextLink href="https://github.com/seek-oss/vocab/">
                      Vocab
                    </TextLink>{' '}
                    ‚Äî SEEK&rsquo;s strongly-typed internationalization framework
                    for React.
                  </Text>
                </Stack>
              </Stack>

              <Divider weight="strong" />

              <Stack space="large">
                <Heading level="2">üèÉ‚Äç‚ôÄÔ∏è Next steps</Heading>
                <Text>
                  Ensure you‚Äôre fully set up with these additional
                  recommendations.
                </Text>
                <Stack space="gutter">
                  <Heading level="3">Brand resources</Heading>
                  <Text>
                    To align your project to SEEK's brand standards it is
                    recommended to install{' '}
                    <TextLink href="https://github.com/SEEK-Jobs/shared-web-assets">
                      Shared web assets
                    </TextLink>
                    .
                  </Text>
                </Stack>

                <Stack space="gutter">
                  <Heading level="3">Telemetry</Heading>
                  <Text>
                    To help us improve sku, please install our private{' '}
                    <TextLink href="https://github.com/SEEK-Jobs/sku-telemetry">
                      sku telemetry
                    </TextLink>{' '}
                    package that gives us insights on usage, errors and
                    performance.
                  </Text>
                </Stack>
              </Stack>

              <Divider weight="strong" />

              <Stack space="large">
                <Heading level="2">üôã‚Äç‚ôÇÔ∏è Support</Heading>

                <Text>
                  Whether it&rsquo;s a problem or a feature request, please
                  don&rsquo;t hesitate to contact us via slack.
                </Text>

                <Stack space="gutter">
                  <Heading level="3">Technical support</Heading>

                  <Text>
                    For technical queries about the platform or the design
                    system:
                  </Text>

                  <List>
                    <Text>
                      <TextLink href="https://seekchat.slack.com/channels/sku-support">
                        #sku-support
                      </TextLink>
                    </Text>
                    <Text>
                      <TextLink href="https://seekchat.slack.com/channels/braid-support">
                        #braid-support
                      </TextLink>
                    </Text>
                  </List>
                </Stack>

                <Stack space="gutter">
                  <Heading level="3">Design support</Heading>

                  <Text>
                    To discuss new or existing design patterns, visit{' '}
                    <TextLink href="https://seekchat.slack.com/channels/braid-design-support">
                      #braid-design-support
                    </TextLink>
                    .
                  </Text>
                </Stack>
              </Stack>

              <Divider weight="strong" />

              <Stack space="large">
                <Heading level="2">üì£ Announcements</Heading>

                <Text>
                  For keeping up to date with new features and bug fixes as they
                  are released, it&rsquo;s recommended to join and turn on
                  notifications for the following channels:
                </Text>

                <List>
                  <Text>
                    <TextLink href="https://seekchat.slack.com/channels/front-end-tooling-announcements">
                      #front-end-tooling-announcements
                    </TextLink>
                  </Text>
                  <Text>
                    <TextLink href="https://seekchat.slack.com/channels/braid-announcements">
                      #braid-announcements
                    </TextLink>
                  </Text>
                </List>
              </Stack>
            </Stack>
          </Box>
        </Stack>
      </ContentBlock>
    </Box>
  );
}
"
`;
